enumerable-method.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # my_each
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
+    end
+    # Analysis:
+    # Self represent the current object, returns the elements of the array
+    # Yield asks for a block
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+  end
+
+  # my_select
+  def my_select
+    # new array
+    select_arr = []
+
+    my_each do |i|
+      select_arr << i if yield(i)
+    end
+    select_arr
+    # select: Returns a new array containing all elements of enum for which the given block returns a true value.
+  end
+
+  # my_all?
+  def my_all?
+    # all: The method returns true if ALL elements are true (or empty array).
+
+    length.times do |i|
+      return false unless yield self[i]
+    end
+    true
+  end
+
+  # my_any?
+  def my_any?
+    # any: The method returns true if AT LEAST one element is true (or non empty array).
+
+    selected = my_select do |i|
+      yield(i)
+    end
+    !selected.empty?
+  end
+
+  # my_none?
+  def my_none?
+    # none: The method returns true if NO elements are true (or empty array).
+
+    selected = my_select { |i| yield(i) }
+    selected.length <= 0
+  end
+
+  # my_count
+  def my_count
+    # count: The count method takes an enumerable collection and counts how many elements match the given criteria.
+    count = 0
+    my_each do |element|
+      count += 1 if !block_given? || yield(element)
+    end
+    count
+  end
```

Comment:
if we input an argument (instead of a block), it should work too, example:

```
irb(main):010:0> arr = [1,2,3]
=> [1, 2, 3]
irb(main):011:0> arr.count(2)
=> 1
```